templates/common/infra/bicep/core/host/aks-managed-cluster.bicep

```diff
@@ -0,0 +1,139 @@
+@description('The name for the AKS managed cluster')
+param name string
+
+@description('The name of the resource group for the managed resources of the AKS cluster')
+param nodeResourceGroupName string = ''
+
+@description('The Azure region/location for the AKS resources')
+param location string = resourceGroup().location
+
+@description('Custom tags to apply to the AKS resources')
+param tags object = {}
+
+@description('Kubernetes Version')
+param kubernetesVersion string = '1.23.12'
```

Comment:
I think we want to move this forward to something newer, as the 1.23 series is leaving support soon

https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar

Should we pick 1.25.X since that's the most recent version?